### DISCUSSION TITLE
Run command once daemon from service is ready

### DISCUSSION THREAD

**Discussion by omnigenous** [davmac314/dinit#411] [Q&A] [answered]
After resolving https://github.com/davmac314/dinit/discussions/410 I noticed that now service in question exits with status 0

It makes sense since first command is starting daemon and second does a thing and exists.

My question is how to keep monitoring daemon (first command) but still run command after daemon is ready (or the way I do it is to wait couple seconds and then run second command)

---

Instead I used created another service and used `depends-on` and that works but now I have second service exited (as expected)

```
[     {-}] second-command (exit status: 0)
```

But it gives wrong impression like something went wrong, should I just accept it?

What is a proper way to run command after daemon from service has started?

**Comment by aanderse** [comment]
maybe it seems like feature bloat but `ExecStartPre` (and sometimes `ExecStartPost`) are nice conveniences to have especially for cases like this...

**Comment by davmac314** [comment, marked as answer]
If you don't want a service to show as stopped once the process terminates, use a `type = scripted` service. (Again, this is covered in the documentation).
